lib/dal/src/code_view.rs

```diff
@@ -49,18 +63,58 @@ pub struct CodeView {
     pub func: Option<String>,
 }
 
+#[derive(Deserialize, Debug)]
+struct CodeGenerationEntry {
+    pub code: Option<String>,
+    pub format: Option<String>,
+    #[serde(default)]
+    pub message: Option<String>,
+}
+
 impl CodeView {
-    pub fn new(
-        language: CodeLanguage,
-        code: Option<String>,
-        message: Option<String>,
-        func: Option<String>,
-    ) -> Self {
-        CodeView {
-            func,
+    pub async fn new(
+        ctx: &DalContext,
+        attribute_value_id: AttributeValueId,
+    ) -> Result<Option<Self>, CodeViewError> {
+        let attribute_value = AttributeValue::get_by_id(ctx, attribute_value_id).await?;
+        let code_view_name = match attribute_value.key(ctx).await? {
+            Some(key) => key,
+            None => return Ok(None),
+        };
+
+        let func_execution = match attribute_value.materialized_view(ctx).await? {
+            Some(func_execution) => func_execution,
+            None => return Ok(None),
+        };
+
+        let code_gen_entry: CodeGenerationEntry = serde_json::from_value(func_execution)?;
+        if code_gen_entry.code.is_none() || code_gen_entry.format.is_none() {
+            return Ok(None);
+        }
+
+        // Safe unwraps because of the above check
+        let format = code_gen_entry.format.as_ref().unwrap();
+        let code = code_gen_entry.code.as_ref().unwrap();
```

Comment:
@stack72 I'd recommend using a match statement here to get that behavior.

```rust
let (code, format) = match (code_gen_entry.code, code_gen_entry.format) {
  (None, None) => return Ok(None),
  (Some, Some) => // do stuff,
  (None, None) => return Err(// new error),
};

Reply:
I can circle back on this and make this better - it's the same logic that existed before so I didn't change it